server/models/database/building_db_model.py

```diff
@@ -32,7 +32,15 @@ async def check_name_exits(cls, name: str) -> bool:
         return await cls.find_one(cls.name == name) is not None
 
     @classmethod
-    async def by_ids(cls, ids: list[str]) -> list[Self]:
+    async def check_name_is_valid(cls, building_id: str, name: str) -> bool:
```

Comment:
então, esse tipo de checagem eu não sei até onde é bom, pois deixa lento o caso de uso comum.
O caso de uso comum é criar um prédio cujo nome ainda não exista.
Da forma como está, sempre que for criar um novo prédio, ele vai terq fazer essa checagem antes.
Talvez seja interessante estudar o erro que o Beanie vai retornar caso um prédio com mesmo nome for criado, já que está indexado únicamente por nome, provavelmente um index error. Baseado nesse erro jogar a exceção HTTP pro usuário

Reply:
De forma mais abrangente, acho que devemos pensar em utilizar melhor as exceções que o banco de dados joga pra gente. Como já estamos indexando o bd, é redundante fazer a checagem antes de tentar inserir